### PR TITLE
APP-59 CC32xx build and out-of-memory fixes

### PR DESCRIPTION
**Pull request by forGGe**
Various improvements. SDK part is pending.

### CODE REVIEW

**Comment by forGGe** [comment]
@dyosick @rasendubi

File: gpiocontrol/source/cc32xx/CMakeLists.txt [hunk 1]
```diff
@@ -12,32 +12,43 @@
...
-  find_package (OpenSSL REQUIRED)
-  set(APP_LIBS kaac_s crypto)
-  set(SAMPLE_SOURCE_FILES
-            src/kaa_demo.c 
-     )
-  link_directories(${CMAKE_CURRENT_SOURCE_DIR}/build)
+# Target-independent flags.
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -Wall -Wextra")
+
+if(KAA_DEBUG_ENABLED)
```
Comment (forGGe) [inline gpiocontrol/source/cc32xx/CMakeLists.txt:25]: Missed `TODO: KAA-858` here

File: gpiocontrol/source/cc32xx/build.sh [hunk 2]
```diff
@@ -48,7 +59,8 @@ echo "==================================="
 echo " ACCESS TOKEN: " $DEMO_ACCESS_TOKEN
 echo "==================================="
 
-function build_thirdparty {
+
+unpack_sdk() {
     if [[ ! -d "$KAA_C_LIB_HEADER_PATH" &&  ! -d "$KAA_CPP_LIB_HEADER_PATH" ]]
```
Comment (rasendubi) [inline gpiocontrol/source/cc32xx/build.sh]: sh doesn't support `[[ ... ]]`

File: gpiocontrol/source/cc32xx/src/kaa_demo.c [hunk 3]
```diff
@@ -105,58 +88,55 @@ void kaa_GPIOToggle_info_request(void *context
 
 int main(/*int argc, char *argv[]*/)
```
Comment (rasendubi) [inline gpiocontrol/source/cc32xx/src/kaa_demo.c]: Make this `(void)`

File: gpiocontrol/source/cc32xx/targets/cc32xx/cc32xx_support.c [hunk 4]
```diff
@@ -187,7 +223,7 @@ void wlan_configure()
 
     sl_WlanSetMode(ROLE_STA);
     sl_WlanPolicySet(SL_POLICY_CONNECTION, SL_CONNECTION_POLICY(1, 0, 0, 0, 1), NULL, 0);
-    sl_WlanProfileDel(0xFF);
+    sl_WlanProfileDel(0xF);
```
Comment (rasendubi) [inline gpiocontrol/source/cc32xx/targets/cc32xx/cc32xx_support.c]: Magic number

File: gpiocontrol/source/cc32xx/CMakeLists.txt (hunk 1 above)
Comment (forGGe) [inline gpiocontrol/source/cc32xx/CMakeLists.txt:25]: Done

File: gpiocontrol/source/cc32xx/build.sh (hunk 2 above)
Comment (forGGe) [inline gpiocontrol/source/cc32xx/build.sh]: Done

File: gpiocontrol/source/cc32xx/src/kaa_demo.c (hunk 3 above)
Comment (forGGe) [inline gpiocontrol/source/cc32xx/src/kaa_demo.c]: Done

File: gpiocontrol/source/cc32xx/targets/cc32xx/cc32xx_support.c (hunk 4 above)
Comment (forGGe) [inline gpiocontrol/source/cc32xx/targets/cc32xx/cc32xx_support.c]: Done

**Comment by forGGe** [comment]
@Deepnekroz

File: gpiocontrol/source/cc32xx/build.sh [hunk 5]
```diff
@@ -59,61 +68,57 @@ function build_thirdparty {
...
-    cd $BUILD_DIR
-    ENV_VAR=" -DKAA_PLATFORM=cc32xx -DAPP_NAME=$APP_NAME"
-        if [ "$(expr substr $(uname -s) 1 9)" == "CYGWIN_NT" ]; then
-                ENV_VAR=$ENV_VAR" -DKAA_TOOLCHAIN_PATH=c:/cygwin/opt/kaa"
-        fi
 
-    cmake -G "Unix Makefiles" -DSSID=$SSID -DPWD=$PASSWORD -DDEMO_ACCESS_TOKEN=$DEMO_ACCESS_TOKEN $ENV_VAR .. &&
+    KAA_TOOLCHAIN_PATH_SDK="-DCMAKE_TOOLCHAIN_FILE=$KAA_LIB_PATH/toolchains/cc32xx.cmake"
+
+    # TODO: comments about KAA_PLATFORM and KAA_TARGET
```
Comment (rasendubi) [inline gpiocontrol/source/cc32xx/build.sh]: TODO

**Comment by rasendubi** [comment]
LGTM

**Pull request merged**